src/core/infer_request.cc

```diff
@@ -159,6 +159,9 @@ InferenceRequest::Release(std::unique_ptr<InferenceRequest>&& request)
     (*it)();
   }
   request->release_callbacks_.clear();
+  // Reset here instead of PrepareForInference() because InferenceRequest users
```

Comment:
I wouldn't reset secondary_stats_aggregator here or elsewhere. It will be up to the code reusing an infer_request object to leave the stats aggregator the same or to change it (or set it to nullptr).